src/main/java/com/byznass/tiolktrack/kernel/handler/PersistLocationHandler.java

```diff
@@ -0,0 +1,44 @@
+package com.byznass.tiolktrack.kernel.handler;
+
+import com.byznass.tiolktrack.kernel.dao.GpsProvider;
+import com.byznass.tiolktrack.kernel.dao.LocationPersister;
+import com.byznass.tiolktrack.kernel.dao.NoGpsWithIdException;
+import com.byznass.tiolktrack.kernel.model.Location;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+
+public class PersistLocationHandler {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(PersistLocationHandler.class);
+
+	private final LocationPersister locationPersister;
+	private final GpsProvider gpsProvider;
+
+	@Inject
+	public PersistLocationHandler(LocationPersister locationPersister, GpsProvider gpsProvider) {
+
+		this.locationPersister = locationPersister;
+		this.gpsProvider = gpsProvider;
+	}
+
+	public Location persist(Location location) {
+
+		//TODO(TT-31) add validation of latitude and longitude
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.tiolktrack.byznass.com/coding_rules#rule_key=squid%3AS1135)